crates/evm/core/src/utils.rs

```diff
@@ -95,6 +95,26 @@ pub fn get_function<'a>(
 /// Accounts for an impersonated transaction by resetting the `env.tx.caller` field to `tx.from`.
 pub fn configure_tx_env(env: &mut EnvMut<'_>, tx: &Transaction<AnyTxEnvelope>) {
     let impersonated_from = is_impersonated_tx(&tx.inner).then_some(tx.from());
+    configure_tx_env_inner(env, &tx.clone(), impersonated_from);
```

Comment:
after reviewing this again, I tink we should actually just always use the rpc response from value here?